lib/test/Test/Pos/Ssc/Toss/BaseSpec.hs

```diff
@@ -616,6 +622,33 @@ checksBadCertsPayload (GoodPayload epoch sgs certsMap mrs) pk cert =
         not (vcVssKey cert `elem` allVssKeys))
        ==> res1 .&&. res2 .&&. res3 .&&. res4
 
+-- [CT-99]: Test that no two stakeholders can have same active VSS keys
+checksBijectiveCertsPayload :: HasConfiguration => GoodCertsPayload -> VssCertData -> Property
+checksBijectiveCertsPayload (GoodPayload _ _ vssCertsMap _) _ = do
+    -- 1. A VSS payload with two certs that have equal VSS keys should not be
+    -- deserializable (with decode).
+
+    -- insert same certificate with different ids
+    let certsMap = getVssCertificatesMap vssCertsMap
+    let x@(_, cert1) : (sid2, _) : rest = HM.toList certsMap
+    let newCerts = UnsafeVssCertificatesMap $ HM.fromList $ x : (sid2, cert1) : rest
+
+    -- check that certificate duplication throws error on decoding stage
+    let encodedCerts = encode newCerts
+    let decodedCerts = deserialiseFromBytes (decode @VssCertificatesMap)
+                                            (toLazyByteString encodedCerts)
+    let check1 = HM.size certsMap >= 2 ==> isLeft decodedCerts
+
+    -- 2. A VSS payload should not be allowed to contain a certificate with a
+    -- VSS key that already exists in VssCertData. It looks like something
+    -- similar is checked in Test.Pos.Ssc.Toss.BaseSpec, but I'm not quite sure.
+    -- It's definitely possible to write a simpler, foolproof test, though.
+    -- TODO: 'checksBadCertsPayload' doesn't have 'VssCertData' in its arguments
+    -- :shrug: I don't really understand what should be here...
+    let check2 = True
```

Comment:
@ChShersh if the `checksBadCertsPayload` test doesn't make sense, it's fine if you refactor or delete it. I wrote it a few months ago and it may no longer make sense as a property.

Reply:
Also, what do you mean here: `'checksBadCertsPayload' doesn't have 'VssCertData' in its arguments`?
As far as I can tell, this test can supersede `checksBadCertsPayload` because it's also using cases with invalid data and checking that the right failures occur.

Reply:
> Also, what do you mean here: 'checksBadCertsPayload' doesn't have 'VssCertData' in its arguments?

It means that comment message says about `VssCertData` but `checksBadCertsPayload` knows nothing about `VssCertData` so I'm not sure how to _only refactor_ this test.